primitives/src/lib.rs

```diff
@@ -32,7 +35,57 @@
 mod commitment;
 pub mod witness;
 
+pub use commitment::{Commitment, SignedCommitment};
+
+use codec::{Codec, Decode, Encode};
+use sp_std::prelude::*;
+
+/// Key type for BEEFY module.
+pub const KEY_TYPE: sp_application_crypto::KeyTypeId = sp_application_crypto::KeyTypeId(*b"beef");
+
+/// BEEFY application-specific crypto types using ECDSA.
+pub mod ecdsa {
+	mod app_ecdsa {
+		use sp_application_crypto::{app_crypto, ecdsa};
+		app_crypto!(ecdsa, crate::KEY_TYPE);
+	}
+
+	sp_application_crypto::with_pair! {
+		/// A BEEFY authority keypair using ECDSA as its crypto.
+		pub type AuthorityPair = app_ecdsa::Pair;
+	}
+
+	/// Identity of a BEEFY authority using ECDSA as its crypto.
+	pub type AuthorityId = app_ecdsa::Public;
+
+	/// Signature for a BEEFY authority using ECDSA as its crypto.
+	pub type AuthoritySignature = app_ecdsa::Signature;
+}
+
+/// The `ConsensusEngineId` of BEEFY.
+pub const BEEFY_ENGINE_ID: sp_runtime::ConsensusEngineId = *b"BEEF";
```

Comment:
is upper case notation used for other consensus engines?

Reply:
Looking at `ConsensusEngineId` in substrate it seems that there's no standard 🙈 Half are uppercase and the other half is lowercase. I don't have a strict opinion on this, can rename it to `beef` :)

Reply:
Yeah, I'm fine with either, was just asking if there is a convention here.